### ISSUE TITLE
Signature incorrect for Coroutine throw?

### ISSUE THREAD

**Issue by alicederyn** [python/typeshed#9586]
I own some code that has been forwarding throw by declaring a three-parameter method with two defaulted to `None`, but as of py3.11 we have started seeing errors:

```text
TypeError: throw() third argument must be a traceback
```

This doesn't seem to match the type stubs. Should the second `Coroutine.throw` overload be changed to remove the final `| None`? https://github.com/python/typeshed/blob/a9b4fa0a325f2806bde895d2cd374d42cd06e67e/stdlib/typing.pyi#L406

----

## Minimal repro

(Must be run in py3.11)

```python
>>> import asyncio
>>> async def f():
...   await asyncio.sleep(0)
>>> async def g():
...   await asyncio.gather(f(), f())
>>> h = g()
>>> h.send(None)
<stdin>:2: DeprecationWarning: There is no current event loop
<_GatheringFuture pending>
>>> h.throw(ValueError("some error"), None, None)
Traceback (most recent call last):
  ...
TypeError: throw() third argument must be a traceback
```

Notes:
* this does not happen if you instead do `h = f()`, nor if you skip the `h.send(None)`, so there is something important about the `asyncio.gather`.
* if you run this in py3.10, you will get `ValueError: some error`, so this is a behaviour change in 3.11.
* Possibly this is caused by https://github.com/python/cpython/commit/0a8b8e0d262eae83ffbc6b414a1f5747cdbd1d88 ?

**Comment by srittau** [comment]
I couldn't find anything about this in the [release notes](https://docs.python.org/3/whatsnew/3.11.html). I suspect this is actually a bug in Python 3.11.

**Comment by hauntsaninja** [comment]
Can confirm that this behaviour is changed by the commit you mention https://github.com/python/cpython/commit/0a8b8e0d262eae83ffbc6b414a1f5747cdbd1d88

**Comment by hauntsaninja** [comment]
Not sure why it was changed, possibly unintentionally since the PR is missing a news entry. Seems easy enough to fix, let me open an issue against CPython...

**Comment by srittau** [comment]
This has now been fixed in cpython main and a backport PR for 3.11 is open. I don't think we need to do anything here. Thanks, @alicederyn for opening the issue and @hauntsaninja for fixing it in CPython!
